src/librustdoc/html/static/js/search.js

```diff
@@ -154,13 +389,101 @@ window.initSearch = function(rawSearchIndex) {
+            elemName: null,
+            args: [],
+            returned: [],
+            // Total number of elements (includes generics).
+            totalElems: 0,
+            // Total number of "top" elements (does not include generics).
+            foundElems: 0,
+            // This field is used to check if it's needed to re-run a search or not.
+            id: "",
+            // This field is used in `sortResults`.
+            nameSplit: null,
+            literalSearch: false,
+            error: null,
+        };
+        query.id = userQuery;
+        try {
+            parseInput(query);
+        } catch (err) {
+            query.error = err.message;
+            query.elems = [];
+            query.returned = [];
+            query.args = [];
+            return query;
+        }
+        query.foundElems = query.elems.length + query.args.length + query.returned.length;
+        if (!query.literalSearch) {
+            // If there is more than one element in the query, we switch to literalSearch in any
+            // case.
+            query.literalSearch = query.foundElems > 1;
+        }
+        if (query.elemName !== null) {
+            query.foundElems += 1;
+        }
+        if (query.foundElems === 0 && userQuery.length !== 0) {
+            // In this case, we'll simply keep whatever was entered by the user...
+            createQueryElement(query, query.elems, userQuery, []);
+            query.foundElems += 1;
+        }
+        if (query.typeFilter !== null) {
+            query.typeFilter = query.typeFilter.replace(/^const$/, "constant");
+            query.typeFilter = itemTypeFromName(query.typeFilter);
+        } else {
+            query.typeFilter = NO_TYPE_FILTER;
+        }
+        // In case we only have one argument, we move it back to `elems` to keep things simple.
+        if (query.foundElems === 1 && query.elemName !== null) {
+            query.elems.push(query.elemName);
+            query.elemName = null;
+        }
+        if (query.elemName !== null || query.elems.length === 1) {
+            userQuery = query.elemName || query.elems[0];
+            query.nameSplit = typeof userQuery.path === "undefined" ? null : userQuery.path;
+        }
+        return query;
+    }
+
+    /**
+     * Creates the query results.
+     *
+     * @param {Array<Object>} results_in_args
+     * @param {Array<Object>} results_returned
+     * @param {Array<Object>} results_in_args
+     * @param {ParsedQuery} queryInfo
+     * @return {Object}                        - A search index of results
+     */
+    function createQueryResults(results_in_args, results_returned, results_others, queryInfo) {
+        return {
+            "in_args": results_in_args,
+            "returned": results_returned,
+            "others": results_others,
+            "query": queryInfo,
+        };
+    }
 
-        removeEmptyStringsFromArray(split);
+    /**
+     * Executes the query and builds an index of results
```

Comment:
The results are not an index. See suggestion above - let's call it a table of results.

Reply:
Indeed, reworded a bit.